invenio_rdm_records/services/pids/providers/datacite.py

```diff
@@ -109,16 +110,21 @@ def __init__(
         self.serializer = serializer or DataCite43JSONSerializer()
 
     @staticmethod
-    def _log_errors(errors):
+    def _log_errors(exception):
         """Log errors from DataCiteError class."""
-        # DataCiteError is a tuple with the errors on the first
-        errors = json.loads(errors.args[0])["errors"]
-        for error in errors:
-            field = error.get("source", None)
+        # DataCiteError will have the response msg as first arg
+        ex_txt = exception.args[0] or ""
+        try:
+            ex_json = json.loads(ex_txt)
+        except JSONDecodeError:
+            current_app.logger.error(f"No content error: {ex_txt}")
+            return
+
+        # the `errors` field is only available when a 4xx error happened (not 500)
+        for error in ex_json.get("errors", []):
+            field = error.get("source", "Missing field/error")
```

Comment:
I'd switch the default text from "Missing field/error" to something like "DataCite service; you may want to check your password". I'm pretty sure that if you have a missing field you'll always get back a source, so it doesn't make sense to have missing field in the default. The most common error I've seen people run into is password issues, since DataCite just returns a plain 403. I'd like to put some indication in the logs that it's something folks should look for., but I'm not sure of the exact phrasing.

Reply:
Thanks a lot for the review and for linking the issue: I should have checked if there was one.
We actually encountered this error because of a 500 Internal server error on Datacite APIs, not because of 403.
The error is caught earlier, thanks to different exceptions [here](https://github.com/inveniosoftware/datacite/blob/master/datacite/errors.py#L38): I will need some refactoring to have a better error msg.

Reply:
@tmorrell I have refactored the func, have a look if now it is more clear. Tests should also clarify the behaviour.

Reply:
Awesome-that's much more readable and should be clearer for folks. It also makes it easier if we have to add more log messages or help text in the future.